### PR TITLE
Subdivision (in the sense of domain decomposition)

### PR DESCRIPTION
**Pull request by timovanopstal**
Parallelization in IFEM is currently patch-based, this PR decouples parallelization from patch structure. In this way geometry generation is not driven by parallelization considerations.

### CODE REVIEW

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
undocumented changes since requesting review:
- fix vtf/u output for subdivision
- add nonhomogeneous Dirichlet test (https://github.com/sintefmath/IFEM-Poisson/issues/8)
- fix LR tests

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

File: src/SIM/TopologySet.h [hunk 1]
```diff
@@ -32,6 +32,9 @@ struct TopItem
   short int idim;  //!< Dimension on the local item [-3,3]
 
   //! \brief Default constructor.
+  //!< \param[in] p Patch index
+  //!< \param[in] i Index of topology item
+  //!< \param[in] d Dimension of topology item
```
Comment (kmokstad) [inline src/SIM/TopologySet.h]: Incorrect doxy, remove the '<'-characters

**Comment by timovanopstal** [comment]
jenkins build this please

**Comment by timovanopstal** [comment]
jenkins build this please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
jenkins build this with downstreams ifem-poisson=8 please

**Comment by timovanopstal** [comment]
merging manually...

**Pull request merged**